.github/workflows/templates-go.yaml

```diff
@@ -0,0 +1,112 @@
+name: Templates
+
+on:
+  push:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/templates-go.yaml"
+      - "templates/**"
+  pull_request:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/templates-go.yaml"
+      - "templates/**"
+
+env:
+  TINYGO_VERSION: "0.33.0"
```

Comment:
I don't believe this is used anywhere, though if we decided to only support the latest stable version of TinyGo, it would seem reasonable to pull the version of TinyGo we currently target from this